language/move-vm/runtime/src/runtime.rs

```diff
@@ -201,24 +204,54 @@ impl<'alloc> VMRuntime<'alloc> {
     }
 }
 
+fn verify_type_args(signature: &FunctionSignature, ty_args: &[Type]) -> VMResult<()> {
+    if signature.type_formals.len() != ty_args.len() {
+        return Err(
+            VMStatus::new(StatusCode::CONTRAINT_KIND_MISMATCH).with_message(format!(
+                "type argument length mismatch: expected {} got {}",
+                signature.type_formals.len(),
+                ty_args.len()
+            )),
+        );
+    }
+    for (k, ty) in signature.type_formals.iter().zip(ty_args.iter()) {
```

Comment:
nit, you don't need the `.iter()` on the argument to `zip`. It takes anything that implements `IntoIterator`
super mega nit, I would switch the order of type_formals and ty_args, so its `(ty, k)` instead of `(k, ty)`